api/api_public_debug.go

```diff
@@ -56,19 +57,24 @@ func (api *PublicDebugAPI) GetBlockRlp(ctx context.Context, number uint64) (stri
 }
 
 // PrintBlock retrieves a block and returns its pretty printed form.
-func (api *PublicDebugAPI) PrintBlock(ctx context.Context, number uint64) (string, error) {
-	block, _ := api.b.BlockByNumber(ctx, rpc.BlockNumber(number))
+func (api *PublicDebugAPI) PrintBlock(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) (string, error) {
+	block, _ := api.b.BlockByNumberOrHash(ctx, blockNrOrHash)
 	if block == nil {
-		return "", fmt.Errorf("block #%d not found", number)
+		blockNumberOrHashString, _ := blockNrOrHash.NumberOrHashString()
+		return "", fmt.Errorf("block %v not found", blockNumberOrHashString)
 	}
 	return spew.Sdump(block), nil
 }
 
 // SeedHash retrieves the seed hash of a block.
-func (api *PublicDebugAPI) SeedHash(ctx context.Context, number uint64) (string, error) {
-	block, _ := api.b.BlockByNumber(ctx, rpc.BlockNumber(number))
+func (api *PublicDebugAPI) SeedHash(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) (string, error) {
```

Comment:
1. The input type is changed to support blockHash but the API doesn't support blockHash.
2. This API is for gxhash(PoW) consensus algorithm which is not maintained. I think we better deprecate this API.

Reply:
1. Oh.. I misunderstood this API as only for blockNumber. It's possible to support blockHash.
```
func (api *PublicDebugAPI) SeedHash(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) (string, error) {	
	block, err := api.b.BlockByNumberOrHash(ctx, blockNr)	
	if err != nil {		
	   blockNumberOrHashString, _ := blockNrOrHash.NumberOrHashString()
	   return "", fmt.Errorf("block %v not found", blockNumberOrHashString)
	}	
	return fmt.Sprintf("0x%x", gxhash.SeedHash(uint64(block.header.number))), nil}
}
```
2. However, as you said, it seems unnecessary API, so I will deprecate this API.

Thanks! Aidan.